src/util/logging/hooks.go

```diff
@@ -0,0 +1,84 @@
+package logging
+
+import (
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+)
+
+func NewReplayHook(logger *Logger) ReplayHook {
+	return ReplayHook{
+		Logger: logger,
+		excludeTypes: map[reflect.Type]int{
+			reflect.TypeOf(ReplayHook{}):    1,
+			reflect.TypeOf(ModuleLogHook{}): 1,
+		},
+	}
+}
+
+// Do not replay hooks of given exclude types
+func ExclusiveReplayHook(logger *Logger, exclude []reflect.Type) (h ReplayHook) {
+	h = NewReplayHook(logger)
+	h.excludeTypes = nil
+	for _, _type := range exclude {
+		h.excludeTypes[_type] = 1
```

Comment:
This will segfault because excludeTypes is nil, must be allocated to a `map[reflect.Type]int`

Reply:
It's fine for slices to be nil https://tour.golang.org/moretypes/12

Reply:
oh this is a map ... u r right